src/uu/wc/src/wc.rs

```diff
@@ -306,29 +306,83 @@ fn word_count_from_reader<T: WordCountable>(
     mut reader: T,
     settings: &Settings,
 ) -> (WordCount, Option<io::Error>) {
-    let only_count_bytes = settings.show_bytes
-        && (!(settings.show_chars
-            || settings.show_lines
-            || settings.show_max_line_length
-            || settings.show_words));
-    if only_count_bytes {
-        let (bytes, error) = count_bytes_fast(&mut reader);
-        return (
-            WordCount {
-                bytes,
-                ..WordCount::default()
-            },
-            error,
-        );
-    }
-
-    // we do not need to decode the byte stream if we're only counting bytes/newlines
-    let decode_chars = settings.show_chars || settings.show_words || settings.show_max_line_length;
-
-    if !decode_chars {
-        return count_bytes_and_lines_fast(&mut reader);
+    match (
+        settings.show_bytes,
+        settings.show_chars,
+        settings.show_lines,
+        settings.show_max_line_length,
+        settings.show_words,
+    ) {
+        (false, false, false, false, false) => (WordCount::default(), None),
+        (true, false, false, false, false) => {
+            // Fast path when only show_bytes is true.
+            let (bytes, error) = count_bytes_fast(&mut reader);
+            return (
```

Comment:
please fix the clippy warning 
ERROR: `cargo clippy`: unneeded `return` statement (file:'src/uu/wc/src/wc.rs', line:320)

Reply:
Done, thanks!